pkg/pool-weighted/contracts/smart/InvestmentPool.sol

```diff
@@ -462,7 +462,14 @@ contract InvestmentPool is BaseWeightedPool, ReentrancyGuard {
         // Since what we're doing is sending out collected management fees, we don't require any BPT in exchange: we
         // simply send those funds over.
         bptAmountIn = 0;
-        amountsOut = new uint256[](_getTotalTokens()); // TODO: set the actual token amounts
+
+        amountsOut = new uint256[](_getTotalTokens());
+        for (uint256 i = 0; i < _getTotalTokens(); ++i) {
+            // We can use unchecked getters and setters as we know the map has the same size (and order!) as the Pool's
+            // tokens.
+            amountsOut[i] = _tokenCollectedManagementFees.unchecked_valueAt(i);
+            _tokenCollectedManagementFees.unchecked_setAt(i, 0);
+        }
```

Comment:
I find a bit weird we are recomputing the manager fees in the callback, should we find a way to re-use what we do in `collectManagementFees` or maybe try to decode it from `userData` here?

Reply:
We have to go over the loop anyway, and I liked having the redudancy provided by both lookups. This is also an infrequent operation, so I'm fine with doing it this way.